spec(consensus/pbts): update description for introduced parameters. (#2206)
cometbft/cometbft: spec/core/data_structures.md

| Height            | uint64                     | Height is the height for this header.                                                                                                                                                                                                                                                                                                                                                 | Must be > 0, >= initialHeight, and == previous Height+1                                                                                                                                          |

> | Time              | [Time](#time)             | The timestamp can be computed using [PBTS][pbts] or [BFT Time][bfttime] algorithms. In case of PBTS, it is the time at which the proposer has produced the block. In case of BFT Time, it is equal to the weighted median of timestamps validators present in the last commit.                                                                                                  | Time must be larger than the Time of the previous block header. The timestamp of the first block should not be smaller than the genesis time. When BFT Time is adopted, it should match the genesis time (since there's no votes to compute the median). |

Reviewer: ```suggestion
| Time              | [Time](#time)             | The timestamp can be computed using [PBTS][pbts] or [BFT Time][bfttime] algorithms. In case of PBTS, it is the time at which the proposer has produced the block. In case of BFT Time, it is equal to the weighted median of timestamps validators present in the last commit.                                                                                                  | Time must be larger than the Time of the previous block header. The timestamp of the first block should not be smaller than the genesis time. When BFT Time is adopted, it should match the genesis time (since there's no votes to compute the median with). |
```

> | Time              | [Time](#time)             | The timestamp can be computed using [PBTS][pbts] or [BFT Time][bfttime] algorithms. In case of PBTS, it is the time at which the proposer has produced the block. In case of BFT Time, it is equal to the weighted median of timestamps validators present in the last commit.                                                                                                  | Time must be larger than the Time of the previous block header. The timestamp of the first block should not be smaller than the genesis time. When BFT Time is used, it should match the genesis time (since there's no votes to compute the median with). |

Reviewer 2: ```suggestion
| Time              | [Time](#time)             | The timestamp can be computed using [PBTS][pbts] or [BFT Time][bfttime] algorithms. In case of PBTS, it is the time at which the proposer has produced the block (the value of its local clock). In case of BFT Time, it is equal to the weighted median of timestamps present in the previous commit.                                                                                                  | Time must be larger than the Time of the previous block header. The timestamp of the first block should not be smaller than the genesis time. When BFT Time is used, it should match the genesis time (since there's no votes to compute the median with). |
```

| Name       | Type                             | Description                                                          | Validation                                                                                               |
|------------|----------------------------------|----------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|

> | Height     | int64                            | Height at which this commit was created.                             | Must be >= 0                                                                                             |

Reviewer: Maybe it's worth adding a note after the table explaining when this field is set to 0 (IIRC, it's on the first block, to signal the absence of signatures)

Author: Right.

Author: Mentioned this below, on the rules for the signatures.

| Name               | Type                            | Description                                                                              | Validation                               |
|--------------------|---------------------------------|------------------------------------------------------------------------------------------|------------------------------------------|
| Type               | [SignedMsgType](#signedmsgtype) | The type of message the vote refers to                                                   | Must be `PrevoteType` or `PrecommitType` |

> | Height             | int64                           | Height for which this vote was created for                                               | Must be >= 0                             |

Reviewer: When is this field set to 0 ?
Looks like it can't be zero, because of [this](https://github.com/cometbft/cometbft/blob/main/types/vote.go#L279)

> | Height             | int64                           | Height for which this vote was created for                                               | Must be >= 0                             |

Author: ```suggestion
| Height             | int64                           | Height for which this vote was created for                                               | Must be > 0                             |
```

| Name      | Type                            | Description                             | Validation                               |
|-----------|---------------------------------|-----------------------------------------|------------------------------------------|
| Type      | [SignedMsgType](#signedmsgtype) | The type of message the vote refers to  | Must be `PrevoteType` or `PrecommitType` |

> | Height    | int64                           | Height in which the vote was provided.  | Must be >= 0                             |

Reviewer: Please double check this change

> | Height    | int64                           | Height in which the vote was provided.  | Must be >= 0                             |

Author: ```suggestion
| Height    | int64                           | Height in which the vote was provided.  | Must be > 0                             |
```

Author: There is not validation for this type. But it can only be produced from votes, for which `Height > 0` is required. Fixing.

| Round     | int64  | Round in which the extension was provided.  | Must be >= 0         |
| ChainID   | string | ID of the blockchain running consensus.     |                      |

## Proposal

Proposal contains height and round for which this proposal is made, BlockID as a unique identifier
of proposed block, timestamp, and POLRound (a so-called Proof-of-Lock (POL) round) that is needed for
termination of the consensus. If POLRound >= 0, then BlockID corresponds to the block that was
or could have been locked in POLRound. The message is signed by the validator private key.

| Name      | Type                            | Description                                                                           | Validation                                              |
|-----------|---------------------------------|---------------------------------------------------------------------------------------|---------------------------------------------------------|
| Type      | [SignedMsgType](#signedmsgtype) | Represents a Proposal [SignedMsgType](#signedmsgtype).                                | Must be `ProposalType`                                  |

> | Height    | uint64                          | Height for which this vote was created for                                            | Must be >= 0                                            |

Author: You are probably right, but the `ValidateBasic()` only rejects when it is negative. Plus, the test unit only considers this case for rejection...

> | Height    | uint64                          | Height for which this vote was created for                                            | Must be >= 0                                            |

Author: Namely, if I change this here, we would have to change also the implementation and tests.

> | Height    | uint64                          | Height for which this vote was created for                                            | Must be >= 0                                            |

Reviewer: Yeah, when I said "but it is a bug"? I was referring to the current status of `ValidateBasic`.
I suggest you leave ">=0" here for the moment, and file an issue to look into `ValidateBasic`, to make sure we don't have a bug there.

|-----------|---------------------------------|---------------------------------------------------------------------------------------|---------------------------------------------------------|
| Type      | [SignedMsgType](#signedmsgtype) | Represents a Proposal [SignedMsgType](#signedmsgtype).                                | Must be `ProposalType`                                  |
| Height    | uint64                          | Height for which this vote was created for                                            | Must be >= 0                                            |
| Round     | int32                           | Round that the commit corresponds to.                                                 | Must be >= 0                                            |
| POLRound  | int64                           | Proof of lock round.                                                                  | Must be >= -1                                           |
| BlockID   | [BlockID](#blockid)             | The blockID of the corresponding block.                                               | [BlockID](#blockid)                                     |

> | Timestamp | [Time](#time)                   | Timestamp represents the time at which the proposer produced the block.               | [Time](#time)                                           |

Reviewer 2: ```suggestion
| Timestamp | [Time](#time)                   | Timestamp represents the time at which the block was produced.               | [Time](#time)                                           |
```
